include/maths/common/CBasicStatistics.h

```diff
@@ -1453,6 +1454,20 @@ std::ostream& operator<<(std::ostream& o,
     return o << CBasicStatistics::print(accumulator);
 }
 
+template<typename T, std::size_t N, typename LESS>
+std::ostream& operator<<(std::ostream& o,
+                         const CBasicStatistics::COrderStatisticsStack<T, N, LESS>& accumulator) {
+    o << core::CScopePrintContainers{} << accumulator;
+    return o;
+}
+
+template<typename T, typename LESS>
+std::ostream& operator<<(std::ostream& o,
+                         const CBasicStatistics::COrderStatisticsHeap<T, LESS>& accumulator) {
+    o << core::CScopePrintContainers{} << accumulator;
+    return o;
+}
```

Comment:
This actually probably doesn't work on reflection. `CScopePrintContainers` checks to see if the type is printable and finds this function, but then it implements the print using this function. So an infinite recursion. It should work if you make these instead:
```suggestion
template<typename T, std::size_t N, typename LESS>
std::ostream& operator<<(std::ostream& o,
                         const CBasicStatistics::COrderStatisticsStack<T, N, LESS>& accumulator) {
    o << core::CContainerPrinter::print(accumulator);
    return o;
}

template<typename T, typename LESS>
std::ostream& operator<<(std::ostream& o,
                         const CBasicStatistics::COrderStatisticsHeap<T, LESS>& accumulator) {
    o << core::CContainerPrinter::print(accumulator);
    return o;
}
```